pkg/ipcache/kvstore.go

```diff
@@ -161,6 +161,7 @@ type IPIdentityWatcher struct {
 	store   storepkg.WatchStore
 	ipcache IPCacher
 
+	clusterName                string
```

Comment:
Why encode cluster name in addition to cluster id?

Reply:
Essentially, when using the *cached prefixes*, the cluster name is appended to the base prefix to isolate the information referring to different clusters. The cluster ID, instead, is not part of the etcd key, nor of the corresponding value.